### ISSUE TITLE
Allow matching to notes not beginning with @

### ISSUE THREAD

**Issue by TakuanPickle** [daeh/zotero-obsidian-citations#12]
Loving the concept behind this plugin, anything that gets the Zotero+Obsidian bond stronger is amazing. Thank you! 

I'm not sure the feasibility of this request, but is it possible to remove the limitation of .md notes starting with a '@'? I'm sure it would allow this plug-in to integrate with a broader array of workflows.

**Comment by daeh** [comment]
totally feasible and fairly easy to implement. I was mostly just putting it off until someone requested it. I'll add it to next release, which I'll hopefully have a chance to work on this week.

**Comment by WhatABloooke** [comment]
Beautiful, thank you so much 😄

**Comment by daeh** [comment]
Hi @TakuanPickle -- let me know how v0.0.14a works for you https://github.com/daeh/zotero-obsidian-citations/releases/tag/v0.0.14a

**Comment by TakuanPickle** [comment]
Hi @daeh It's working a

**Comment by TakuanPickle** [comment]
Hi @daeh 

It's working a lot better, finds files in my vault without the @ now :) thank you for implementing that! 

I'm getting this error when I go to use the 'Open ObsidianMD Note' function though:
![image](https://user-images.githubusercontent.com/74401804/153342320-12868761-b2c8-4e04-81a4-0083487fa972.png)

My current settings:
![image](https://user-images.githubusercontent.com/74401804/153342413-984f6af9-4746-40e0-aed6-891f558adefd.png)

Any help would be great.

**Comment by daeh** [comment]
I bet that supplying a vault ID will fix this. It's easy to implement, but it's odd that it seems to work for most people without the vault param, but for you it throws an error. So two things would be useful:

1) If you copy the URI for that note from within Obsidian, how does it compare to URI that throws an error?
<img width="323" alt="CleanShot 2022-02-10 at 00 36 58@2x" src="https://user-images.githubusercontent.com/12652166/153344666-e52e98c2-750b-465d-85d1-3315df5353cd.png">

2) Can you tell me a bit about your vault structure. How many vaults do you have? Is the vault that contains that reading notes the vault that's open and active when you try to open the MD file from Zotero?

**Comment by TakuanPickle** [comment]
It is odd! 

So I turned off any URI related plug-in and have run some tests on a note that is just in the root folder of my vault. The error still persists.

URI from Obsidian is: `obsidian://open?vault=Takuan&file=levi-strauss1973_HoneyAshes`
URI from the error is: `obsidian://open/?path=C%3A%5Clevi-strauss1973_HoneyAshes.md`

Would it be anything to do with Obsidian Sync? I'm not sure if that gives the vault a new path or anything. I have a couple of vaults, but the one with the reading note is the only one open when I try and open the note from Zotero.

**Comment by daeh** [comment]
looks like there's an errant `/` in the URI that's causing an error. It should be `obsidian://open?` not `obsidian://open/?`. That's very weird, I'm not sure where the `/` is coming from.

While I'm looking into it, will you test if this link works? 
(a) `obsidian://open?path=C%3A%5Clevi-strauss1973_HoneyAshes.md`
and if it doesn't, test if this one does: 
(b) `obsidian://open?vault=Takuan&path=C%3A%5Clevi-strauss1973_HoneyAshes.md`

**Comment by TakuanPickle** [comment]
super weird! I'm on Windows 10:

```
Edition	Windows 10 Home
Version	21H1
Installed on	‎4/‎01/‎2021
OS build	19043.1466
Experience	Windows Feature Experience Pack 120.2212.3920.0
```

Obsidian version v0.13.24 (insider).

As I've set my ZotObsCite folder to my vault root the debug file is both huge and contains personal information. Is there anything I can find in there for you? This is the entry for the example note we're looking at:
```
{
   "citekey": "levi-strauss1973_HoneyAshes",
   "citekey_metadata": "levi-strauss1973_HoneyAshes",
   "citekey_title": null,
   "zotkeys": [],
   "zotids": [],
   "name": "levi-strauss1973_HoneyAshes",
   "path": "C:\\takuan\\levi-strauss1973_HoneyAshes.md"
  },
```
and everything before the vault scan:
```
{
 "version": "0.0.13",
 "zoteroVersion": "5.0.97-beta.65+15ae72d14",
 "isMac": false,
 "isLinux": false,
 "isWin": true,
 "prefs": {
  "matchstrategy": "bbtcitekey",
  "source_dir": "C:\\takuan\\",
  "filefilterstrategy": "customfileregex",
  "filepattern": ".+\\.md",
  "zotkeyregex": "^.Local Library::.+\\/items\\/(\\w+)\\)",
  "metadatakeyword": "citekey"
 },
 "config": {
  "matchstrategy": "bbtcitekey",
  "source_dir": "C:\\takuan",
  "filefilterstrategy": "customfileregex",
  "filepattern": ".+\\.md",
  "zotkeyregex": "^.Local Library::.+\\/items\\/(\\w+)\\)",
  "metadatakeyword": "citekey",
  "bbtinstalled": [
   true
  ],
  "checkSetting": true
 },
```

Neither of those two URIs work. The only one that it likes is the one that Obsidian generated itself:
`obsidian://open?vault=Takuan&file=levi-strauss1973_HoneyAshes`
or
`obsidian://open?vault=Takuan&file=levi-strauss1973_HoneyAshes.md`

**Comment by TakuanPickle** [comment]
Oh, so I moved the file since the first post because the URL was getting cut off! I just thought it would be easier to see and debug if it was in the root folder. Sorry if that wasn't clear!

actual path is `C:\takuan\levi-strauss1973_HoneyAshes.md`. 
The URI I get from Obsidian is `obsidian://open?vault=Takuan&file=levi-strauss1973_HoneyAshes`. 

I just realised that I copied out the error URI wrong! `obsidian://open/?path=C%3A%5Ctakuan%5Clevi-strauss1973_HoneyAshes.md` is the URI for the error. Hope that clears it up!

**Comment by daeh** [comment]
cool thanks. Does this URI work?

`obsidian://open?vault=Takuan&path=C%3A%5Ctakuan%5Clevi-strauss1973_HoneyAshes.md`

**Comment by TakuanPickle** [comment]
no, it doesn't, same 'unable to find vault' error :(

**Comment by daeh** [comment]
ok, i'll add an option to use the `file=` param rather than the `path=` param. No idea why you're having issues -- there are a number of things here that seem weird, but it looks like the `file=` uri should work for you.

**Comment by TakuanPickle** [comment]
okay, hopefully that will work! thanks for your time puzzling this out with me.

**Comment by daeh** [comment]
@TakuanPickle please give `v0.0.15a` a try and let me know if the advanced option to use the 'file' URI works for you https://github.com/daeh/zotero-obsidian-citations/releases/download/v0.0.15a/zotero-obsidian-citations-0.0.15a.xpi

**Comment by daeh** [comment]
closed by 67e2fa298e9832eae744c312a294776c15274b29 but please reopen if there are issues
